### PR TITLE
[Frontend] Distinguish learnable param and non learnable

### PR DESCRIPTION
**Pull request by zhen-jia**
## Description ##
(Brief description on what this PR is about)

## Checklist ##

- [x] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented

cc @awslabs/raf-reviewer

### CODE REVIEW

File: python/raf/optim/sgd.py [hunk 1]
```diff
@@ -124,7 +124,7 @@ def build(self, model):
                     # TODO(@anijain2305): We might mark requires_grad to non-float parameters
                     # which is incorrect. We should improve the gradient pass for better
                     # has_grad results.
-                    if param.requires_grad and "float" in param.dtype:
+                    if param.requires_grad:
```
Comment (comaniac) [inline python/raf/optim/sgd.py:127]: You could now remove the above TODO added by Animesh.

**Review by comaniac — COMMENTED** [review]
Overall LGTM. Could we have a unit test?
Also cc @hzfan @yzhliu @hgt312

**Comment by zhen-jia** [comment]
OK, I will add a unit test later.

**Review by comaniac — APPROVED** [review]
LGTM. cc @hzfan @yzhliu

**Review by yzhliu — APPROVED** [review]

**Comment by comaniac** [comment]
Thanks @zhen-jia @yzhliu

**Pull request merged**